rero_ils/modules/stats/api/librarian.py

```diff
@@ -0,0 +1,312 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2023 RERO
+# Copyright (C) 2019-2023 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""To compute the statistics for librarian."""
+
+import hashlib
+
+import arrow
+from dateutil.relativedelta import relativedelta
+from invenio_search.api import RecordsSearch
+
+from rero_ils.modules.items.api import ItemsSearch
+from rero_ils.modules.items.models import ItemCirculationAction
+from rero_ils.modules.locations.api import LocationsSearch
+from rero_ils.modules.operation_logs.api import OperationLog
+from rero_ils.modules.patrons.api import PatronsSearch
+
+from .pricing import StatsForPricing
+
+
+class StatsForLibrarian(StatsForPricing):
+    """Statistics for librarian.
+
+    TODO: the type of the statistic should be changed from
+          librarian to library.
+    """
+
+    def __init__(self, to_date=None):
+        """Constructor.
+
+        :param to_date: end date of the statistics date range
+        """
+        self.to_date = to_date or arrow.utcnow() - relativedelta(days=1)
+        # Get statistics per month
+        _from = f'{self.to_date.year}-{self.to_date.month:02d}-01T00:00:00'
+        _to = self.to_date.format(fmt='YYYY-MM-DDT23:59:59')
+        self.date_range = {'gte': _from, 'lte': _to}
```

Comment:
Seems not necessary because it's entirely based on `self.to_date`. If `to_date` change, you need to change `date_range` too.